docs/cvl/methods.md

```diff
@@ -178,6 +190,66 @@ shown in the web report can indicate whether a summary was applied.
+The dispatch list will contain a list of patterns and the default summary to use in case no function matched the selector.
+The possible patterns are:
+1. Exact function - a pattern specifying both a contract, and the function signature.
+   Example: `C.foo(uint)`
+2. Wildcard contract - a pattern specifying the function signature to match this signature on all available contracts (including the primary contract).
+   Example: `_.bar(address)`
+3. Wildcard function - a pattern specifying a contract, and matches all external functions in specified contract.
+For the default summary the user can choose one of: `HAVOC_ALL`, `HAVOC_ECF`, `NONDET`.
+The example entry at the head of this section will specify three functions to route calls to:
+1. `C.foo(uint)`
+2. `Other.bar(address)`
+3. `C.baz(bool)`
+With the default being `NONDET`.
+
+Entry annotations ({ref}`envfree`, {ref}`optional`) and the `returns` clause are not allowed on an unresolved-calls entry. 
+Also, the visibility is always external, and no policy should be specified.
+
+For an unresolved function call being summarized with the dispatch list above, the Prover will replace the call with a dynamic resolution of the function call. That is something in the lines of:
+```solidity
+function summarized(address a, bytes calldata data) external {
+  if (uint32(data[0:4]) == 0x11111111 && address == address(c)) {
+    // Function selector was equal to foo's
+    // Call C.foo(...)
+  } else if (uint32(data[0:4]) == 0x22222222 && address == address(o)) {
+    // Function selector was equal to bar's
+    // Call O.bar(...)
+  } else if (uint32(data[0:4]) == 0x33333333 && address == address(c)) {
+    // Function selector was equal to baz's
+    // Call C.baz(...)
+  } else {
+    // Use the default summary which is, in this case, NONDET.
+  }
+}
+```
+
+The dispatch list summary will create a dynamic resolution process that determines the specific function to call at runtime based on the function signature and the target contract address. In the provided example, when an unresolved function call is encountered, the Prover dynamically resolves it by inspecting the function selector in the transaction data and the target contract address. By comparing the function selector against known signatures and verifying the contract address, the Prover identifies the appropriate function to call. This dynamic resolution mechanism is crucial for refining specifications because it enables the Prover to accurately model the behavior of smart contracts, even when the exact function being called is not known statically. By replacing unresolved calls with dynamically resolved calls in the dispatch list summary, the specification becomes more precise, leading to more accurate verification results and improved assurance in the correctness of the smart contract.
```

Comment:
The sentence starting with `This dynamic resolution` should be in a new paragraph

Reply:
Ok